packages/theme/composables/useApi/index.ts

```
@@ -1,7 +1,24 @@
-import { useContext } from '@nuxtjs/composition-api';
+import { useContext, ref } from '@nuxtjs/composition-api';
```

In general. I think that API client do not have to utilize composition api so it will not share more technical debt than necessary. Eg. loading or errors as ref are not really required. Loading can be easily handled by the client and error can be simply thrown so the client can handle it. This is not a big thing maybe, just the less debt risk the better.

---

Good point,

so let me understand correctly that:
1. Instead of a return error that is a Ref, return a plain error? 
2. loading field is not necessary here, we remove it, and components, when useApi is used, takes care of handle loading state?

---

@bartoszherba  In addition to point 1. Maybe we can just throw an error instead of returning an error field? cc: @sethidden

---

If error and loading weren't refs then using useApi would be no different than axios.get()

e.g. loading is a good example - why litter the template with const isXLoading = ref(false), const isYLoading = ref(false) if that can be returned from an useApi? Perhaps Bartek is talking about a different form of useApi (I'm talking about the "simple promise helper" form)